common/src/main/java/net/opentsdb/utils/PluginLoader.java

```diff
@@ -395,15 +410,43 @@ private static void addFile(File f) throws IOException, SecurityException,
    * @throws InvocationTargetException if there is an issue loading the jar
    */
   private static void addURL(final URL url) throws SecurityException, 
-  NoSuchMethodException, IllegalArgumentException, IllegalAccessException, 
-  InvocationTargetException {
-    //URLClassLoader sysloader = new URLClassLoader(new URL[] { url });
-    URLClassLoader sysloader = (URLClassLoader)ClassLoader.getSystemClassLoader();
-    Class<?> sysclass = URLClassLoader.class;
+    NoSuchMethodException, IllegalArgumentException, IllegalAccessException, 
+    InvocationTargetException {
+    
+    final ClassLoader class_loader;
+    if (INSTRUMENTATION != null) {
+      try {
+        INSTRUMENTATION.appendToSystemClassLoaderSearch(new JarFile(url.getFile()));
+        LOG.debug("Successfully added JAR to class loader: " + url.getFile());
+      } catch (IOException e) {
+        throw new IllegalArgumentException("Failed to load JAR: " + url, e);
+      }
+      return;
+    } else if (ClassLoader.getSystemClassLoader() instanceof URLClassLoader) {
+      // Yay! The old pre-JDK9 hack will most likely work      
+      class_loader = ClassLoader.getSystemClassLoader();
+    } else if (Thread.currentThread().getContextClassLoader() instanceof URLClassLoader) {
```

Comment:
Ideally, these last two branches would just be merged into a big error message and an `exit()` call, but for test-case purposes, I'm fine with having them for now.